[DataGrid] Add minHeight to the rendering zone

m4theushw opened the pull request:
Fixes #2133 

Preview: https://deploy-preview-2159--material-ui-x.netlify.app/components/data-grid/components/

oliviertassinari reviewed (commented):
Did you consider moving the fix outside of the virtualization? I mean, AFAIK, this current fix is coupled to it (`renderingZone`). We would face the same issue if we disable the virtualization.

```diff
diff --git a/packages/grid/_modules_/grid/components/containers/GridDataContainer.tsx b/packages/grid/_modules_/grid/components/containers/GridDataContainer.tsx
index a2a6732e..19718d93 100644
--- a/packages/grid/_modules_/grid/components/containers/GridDataContainer.tsx
+++ b/packages/grid/_modules_/grid/components/containers/GridDataContainer.tsx
@@ -17,6 +17,7 @@ export function GridDataContainer(props: GridDataContainerProps) {

   const style: any = {
     minWidth: dataContainerSizes?.width,
+    minHeight: 1,
   };

   if (scrollBar.hasScrollY) {
```

flaviendelangle commented:
https://www.argos-ci.com/mui-org/material-ui-x/builds/2415

Their are scrollbar appearing on several snapshots.

oliviertassinari commented:
@m4theushw The changes surface how the dataContainerSizes.height state is wrong in https://deploy-preview-2159--material-ui-x.netlify.app/components/data-grid/events/#subscribing-to-events:

<img width="627" alt="Capture d’écran 2021-07-21 à 23 08 33" src="https://user-images.githubusercontent.com/3165635/126560393-c5a40a72-a78a-4571-b4d0-c3e5f7eda5c2.png">

But I don't think that we should take care of it now, it was already wrong before (even if we couldn't see the symptoms).

m4theushw commented:
> @m4theushw The changes surface how the dataContainerSizes.height state is wrong in https://deploy-preview-2159--material-ui-x.netlify.app/components/data-grid/events/#subscribing-to-events:

@oliviertassinari I fixed this in da6786c. As consequence, it also removed the empty space after the last row when not using virtualization. When virtualization is enabled, the scrollbar size is not added to the height:

https://github.com/mui-org/material-ui-x/blob/9613bdd0611fa43a53561421f7c783f32769c5dd/packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts#L179-L183

Before: 

<img width="500" src="https://user-images.githubusercontent.com/42154031/126566633-fd46d3d2-789f-4922-bccd-87156c4d316f.png" alt="image" style="max-width:100%;">

After:

<img width="500" src="https://user-images.githubusercontent.com/42154031/126566768-f7dafbcf-fe79-452e-ad98-6a085631b869.png" alt="image" style="max-width:100%;">

oliviertassinari commented:
We get a regression: 

<img width="988" alt="Capture d’écran 2021-07-22 à 00 17 46" src="https://user-images.githubusercontent.com/3165635/126567218-a2d3a938-07d1-4a6b-885d-dbb8c9ab9f90.png">

It looks like we miss a functional test case for it.

m4theushw commented:
> There are scrollbar appearing on several snapshots.

@flaviendelangle Yes, the bug was the scrollbar not appearing. 😁

oliviertassinari commented:
https://github.com/mui-org/material-ui-x/pull/2159#issuecomment-884535893

Exactly why functional test with visual test doesn't work 😆

m4theushw commented:
> We get a regression:

@oliviertassinari Fixed in 033ec03. Now we have a good reason to add the scrollbar size: if `autoHeight` is true. 

From what I can see all other visual test failures are related to https://github.com/mui-org/material-ui-x/pull/2159#issuecomment-884533906. If there's another regression we will discover later.

> Loking at the code, it looks like the abstraction in the state is wrong. The height values aren't meant to be the same. From what I understand, the scroll container should use totalSizes (account for the horizontal scrollbar height if needed), while the scrollable element dataContainerSizes (only account for the space needed to render the children, it's not a minHeight, it's a height ⚠️⚠️⚠️).

Yeah, doesn't make sense to replicate the values. That's something to be solved in another PR.

oliviertassinari commented on packages/grid/_modules_/grid/components/containers/GridDataContainer.tsx:
```diff
@@ -13,15 +10,11 @@ export function GridDataContainer(props: GridDataContainerProps) {
-  const scrollBar = useGridSelector(apiRef!, gridScrollBarSizeSelector);
 
   const style: any = {
     minWidth: dataContainerSizes?.width,
+    minHeight: dataContainerSizes?.height,
```
We should be able to force the dimension. From what I understand, using min allows for the value to go beyond what should never happen (hiding root issues).

```suggestion
    // TODO remove min
    minWidth: dataContainerSizes?.width,
    minHeight: dataContainerSizes?.height,
```

m4theushw replied:
Yes. it was always "min". We can try to set the exact dimension in another PR. If it doesn't work it's because `dataContainerSizes.height` is wrong at some point.

oliviertassinari reviewed (commented):
Great to have more test cases, they move us one step in the right direction.

flaviendelangle commented on packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx:
```diff
@@ -519,15 +519,41 @@ describe('<DataGrid /> - Layout & Warnings', () => {
+      it('should have the correct intrinsic height', () => {
+        render(
+          <div style={{ width: 300 }}>
+            <DataGrid {...baselineProps} headerHeight={40} rowHeight={30} autoHeight />
```
We could extract `headerHeight` into its own variable for readability in the total height formula
